lib/taiko.js

```diff
@@ -1522,6 +1522,38 @@ function textField(attrValuePairs, ...args) {
     };
 }
 
+/**
+ * @example
+ * tap('Gmail')
+ * tap(link('Gmail'))
+ * 
+ * @param {...relativeSelector} args - Proximity selectors
+ * @returns {Promise<Object>} - Object with the description of the action performed.
+ */
+module.exports.tap = async (selector, ...args) => {
+    let elems = await handleRelativeSearch(await elements(selector), args);
+    let elemsLength = elems.length;
+    let isElemAtPoint;
+    for (let elem of elems) {
+        isElemAtPoint = false;
+        await scrollTo(elem);
+        isElemAtPoint = await checkIfElementIsCovered(elem, elems, isElemAtPoint);
+        if (isElemAtPoint) {
+            const type = (await evaluate(elem, function getType() { return this.type; })).value;
+            assertType(elem, () => type !== 'file', 'Unsupported operation, use `attach` on file input field');
+            if (headful) await highlightElemOnAction(elem);
+            const { x, y } = await domHandler.boundingBoxCenter(elem);
+            await inputHandler.tap(x,y);
```

Comment:
This can done like,
```
await doActionAwaitingNavigation(options,async () => {
   await inputHandler.tap(x,y);
}); 
```

not sure why you feel it won't hold good.

Reply:
Fails - TypeError: action is not a function
    at doActionAwaitingNavigation (/Users/saikrisv/git/taiko/lib/taiko.js:2057:11)
    at module.exports.tap (/Users/saikrisv/git/taiko/lib/taiko.js:1545:19)
    at <anonymous>

Reply:
I could see options is missed, first arg for `doActionAwaitingNavigation` is options second is the callback function. Please refer other actions for setting options